pkg/controller/webhooks/postgresql.go

```diff
@@ -0,0 +1,199 @@
+// Unfortunately the webhook generator is not very configurable yet and some
+// values cannot be changed. This will clash if/when more webhooks are generated.
+
+// Generate the webhook manifest from the marker comment
+// //go:generate go run -tags generate sigs.k8s.io/controller-tools/cmd/controller-gen webhook paths=./... output:dir=../../../config/controller/pg-webhooks
+
+// See https://book.kubebuilder.io/reference/markers/webhook for docs
+//+kubebuilder:webhook:verbs=create;update;delete,path=/validate-vshn-appcat-vshn-io-v1-vshnpostgresql,mutating=false,failurePolicy=fail,groups=vshn.appcat.vshn.io,resources=postgresqls,versions=v1,name=postgresql.vshn.appcat.vshn.io,sideEffects=None,admissionReviewVersions=v1
+
+var (
+	pgGK = schema.GroupKind{Group: "vshn.appcat.vshn.io", Kind: "VSHNPostgreSQL"}
+	pgGR = schema.GroupResource{Group: pgGK.Group, Resource: "vshnpostgresqls"}
+)
+
+var _ webhook.CustomValidator = &PostgreSQLWebhookHandler{}
+
+// PostgreSQLWebhookHandler handles all quota webhooks concerning postgresql by vshn.
+type PostgreSQLWebhookHandler struct {
+	client    client.Client
+	log       logr.Logger
+	withQuota bool
+}
+
+// SetupPostgreSQLWebhookHandlerWithManager registers the validation webhook with the manager.
+func SetupPostgreSQLWebhookHandlerWithManager(mgr ctrl.Manager, withQuota bool) error {
+	if !viper.IsSet("PLANS_NAMESPACE") && withQuota {
+		return fmt.Errorf("PLANS_NAMEPSACE env variable needs to be set for quota support")
+	}
+
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(&vshnv1.VSHNPostgreSQL{}).
+		WithValidator(&PostgreSQLWebhookHandler{
+			client:    mgr.GetClient(),
+			log:       mgr.GetLogger().WithName("webhook").WithName("postgresql"),
+			withQuota: withQuota,
+		}).
+		Complete()
+}
+
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type
+func (p *PostgreSQLWebhookHandler) ValidateCreate(ctx context.Context, obj runtime.Object) error {
+
+	pg, ok := obj.(*vshnv1.VSHNPostgreSQL)
+	if !ok {
+		return fmt.Errorf("Provided manifest is not a valid VSHNPostgreSQL object")
+	}
+
+	if p.withQuota {
+		err := p.checkPostgreSQLQuotas(ctx, pg, true)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// ValidateUpdate implements webhook.CustomValidator so a webhook will be registered for the type
+func (p *PostgreSQLWebhookHandler) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) error {
+
+	pg, ok := newObj.(*vshnv1.VSHNPostgreSQL)
+	if !ok {
+		return fmt.Errorf("Provided manifest is not a valid VSHNPostgreSQL object")
+	}
+
+	if p.withQuota {
+		err := p.checkPostgreSQLQuotas(ctx, pg, false)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// ValidateDelete implements webhook.CustomValidator so a webhook will be registered for the type
+func (p *PostgreSQLWebhookHandler) ValidateDelete(ctx context.Context, obj runtime.Object) error {
+	// NOOP for now
```

Comment:
If we don't validate DELETE, then why did You register DELETE in webhook configuration?, reference: config/controller/pg-webhook.yaml

Reply:
My train of thought was:

We need to implement the method anyway to satisfy the interface. So if the webhook is already configured, then we don't have to bother with that if/when we need to validate deletions in the future.

Reply:
fine for me